cluster_management/src/main/java/com/pinterest/rocksplicator/ConfigGenerator.java

```diff
@@ -115,6 +129,9 @@ private void generateShardConfig() {
       }
 
       ExternalView externalView = admin.getResourceExternalView(clusterName, resource);
+      if (externalView == null) {
```

Comment:
if externalView is null.... there is no need to continue for that resource.

This is a possibility that externalView is null in some cases, where there is a race condition between getting all resources and iterating to get externalView for a resource when one resource get's deleted in-between

Reply:
Nice. This is very likely. I will add a continue to skip when externalView is null then and report the stats as well.

Reply:
Also it will be better to log the resource name for which externalView is found to be null, in case investigation is required to be done later.

Reply:
@rajathprasad @gopalrajpurohit  is this too risky to continue generate ShardMap when externalView for certain resource are empty?
Pros: it could skip those situation mentioned by Gopal above, ie. resource got deleted between "getResources" and "resource iteration"
Cons: if some transient getExternalView error returned Null, we would still generate a new shardMap and excluding this resource and update the new shardMap to routers. Then, all requests to the resource will fail.